controller/src/main/java/io/pravega/controller/store/index/ZKHostIndex.java

```diff
@@ -93,12 +94,13 @@ public ZKHostIndex(CuratorFramework client, String hostRoot, Executor executor)
     @Override
     public CompletableFuture<List<String>> getEntities(final String hostId) {
         Preconditions.checkNotNull(hostId);
-        return getChildren(getHostPath(hostId));
+        String hostPath = getHostPath(hostId);
+        return sync(hostPath).thenCompose(v -> getChildren(hostPath));
```

Comment:
Is it really necessary to sync here?

Reply:
Yes, when we get a host to failover, we get its entities and then failover those indexed entities. In case we get stale list of children then we will miss failing over some entities and they will be failed over only when a new controller instance bootstraps.